test/ServiceBus.Tests/SlowConsumingTests/EHSlowConsumingTests.cs

```diff
@@ -0,0 +1,178 @@
+﻿using Microsoft.WindowsAzure.Storage.Table;
+using Orleans;
+using Orleans.AzureUtils;
+using Orleans.Providers.Streams.Common;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+using Orleans.ServiceBus.Providers;
+using Orleans.Storage;
+using Orleans.Streams;
+using Orleans.TestingHost;
+using Orleans.TestingHost.Utils;
+using ServiceBus.Tests.TestStreamProviders;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TestExtensions;
+using UnitTests.GrainInterfaces;
+using UnitTests.Grains.ProgrammaticSubscribe;
+using Xunit;
+
+namespace ServiceBus.Tests.SlowConsumingTests
+{
+    [Collection(TestEnvironmentFixture.DefaultCollection)]
+    public class EHSlowConsumingTests : OrleansTestingBase, IClassFixture<EHSlowConsumingTests.Fixture>
+    {
+        private const string StreamProviderName = "EventHubStreamProvider";
+        private const string StreamNamespace = "EHTestsNamespace";
+        private const string EHPath = "ehorleanstest";
+        private const string EHConsumerGroup = "orleansnightly";
+        private const string EHCheckpointTable = "ehcheckpoint";
+        private static readonly string CheckpointNamespace = Guid.NewGuid().ToString();
+        private static readonly TimeSpan monitorCheckperiod = TimeSpan.FromSeconds(30);
+        private static readonly TimeSpan timeout = TimeSpan.FromSeconds(30);
+        private const double flowControlThredhold = 0.6;
+        public static readonly EventHubStreamProviderSettings ProviderSettings =
+            new EventHubStreamProviderSettings(StreamProviderName);
+
+        private static readonly Lazy<EventHubSettings> EventHubConfig = new Lazy<EventHubSettings>(() =>
+            new EventHubSettings(
+                TestDefaultConfiguration.EventHubConnectionString,
+                EHConsumerGroup, EHPath));
+
+        private static readonly EventHubCheckpointerSettings CheckpointerSettings =
+            new EventHubCheckpointerSettings(TestDefaultConfiguration.DataConnectionString,
+                EHCheckpointTable, CheckpointNamespace, TimeSpan.FromSeconds(1));
+
+
+        private readonly Fixture fixture;
+
+        public class Fixture : BaseTestClusterFixture
```

Comment:
You should override CheckPreconditionsOrThrow and throw SkipException if the connection string is null.
Then in the test class' constructor you must call fixture.EnsurePreconditionsMet()
Also mark all of the tests with SkippableFact.

See usages of EnsurePreconditionsMet if you want some examples.